### ISSUE TITLE
Ensemble method sketch

### ISSUE THREAD

**Issue by yvesx** [tambetm/allenAI#3]
Hi Tambet,
The ensembling part seems working very well. Would you care to describe the ensemble method you used in combining the deep models and the retrieval models in a bit more detail (Bayesian model averaging, Boosting, etc.)?

**Comment by tambetm** [comment]
@tanelp, maybe you can elaborate on this.

**Comment by tanelp** [comment]
Firstly, we ensembled deep models and information retrieval models separately by just averaging the scores weighted by the validation set accuracies. Then, we engineered some additional features (such as answer length etc.), and learnt xgboost model to predict the probability that a particular answer is correct. In summary, I think this is called stacking.

**Comment by yvesx** [comment]
thanks tanelp!
Suppose I'm averaging two models weighted by their validation acc: model A gets 90% validation acc. and model B gets 80% validation acc. Your averaging scheme assigned 0.9/(0.9+0.8)=0.529 weight to model A and 0.8/(0.9+0.8)=0.471 to model B. 
Is that right?

**Comment by tanelp** [comment]
Yes, this is correct.

**Comment by yvesx** [comment]
Thank you!
